sphere-models/src/it/java/io/sphere/sdk/products/search/ProductProjectionSearchModelIntegrationTest.java

```diff
@@ -1,150 +1,49 @@
-    protected static final LocalizedEnumValue LOC_ENUM_TWO = LocalizedEnumValue.of("two-key", LocalizedString.of(GERMAN, "zwei", FRENCH, "deux"));
-    protected static final LocalizedEnumValue LOC_ENUM_THREE = LocalizedEnumValue.of("three-key", LocalizedString.of(GERMAN, "drei", FRENCH, "trois"));
-    protected static final BigDecimal NUMBER_5 = valueOf(5D);
-    protected static final BigDecimal NUMBER_10 = valueOf(10D);
-    protected static final MonetaryAmount MONEY_500_EUR = MoneyImpl.of(valueOf(500), "EUR");
-    protected static final MonetaryAmount MONEY_1000_USD = MoneyImpl.of(valueOf(1000), "USD");
-    protected static final LocalDate DATE_2001 = LocalDate.parse("2001-09-11");
-    protected static final LocalDate DATE_2002 = LocalDate.parse("2002-10-12");
-    protected static final LocalTime TIME_22H = LocalTime.parse("22:05:09.203");
-    protected static final LocalTime TIME_23H = LocalTime.parse("23:06:10.204");
-    protected static final ZonedDateTime DATE_TIME_2001_22H = ZonedDateTime.parse("2001-09-11T22:05:09.203+00:00");
-    protected static final ZonedDateTime DATE_TIME_2002_23H = ZonedDateTime.parse("2002-10-12T23:06:10.204+00:00");
 
     @Rule
     public RetryIntegrationTest retry = new RetryIntegrationTest(10, 10000, LoggerFactory.getLogger(this.getClass()));
 
     @BeforeClass
     public static void setupProducts() {
-        productType = client().executeBlocking(ProductTypeQuery.of().byName(PRODUCT_TYPE_NAME)).head()
-                .orElseGet(() -> createProductType());
-
-        final Query<Product> query = ProductQuery.of()
-                .withPredicates(m -> m.masterData().staged().masterVariant().sku().isIn(asList(SKU1, SKU2, SKU_SOME_ID, SKU_OTHER_ID)));
-        final List<Product> products = client().executeBlocking(query).getResults();
-
-        final Function<String, Optional<Product>> findBySku =
-                sku -> products.stream().filter(p -> sku.equals(p.getMasterData().getStaged().getMasterVariant().getSku())).findFirst();
+        data = createScenario(client());
+        productType = data.getProductType();
+        productA = data.getReferencedProductA();
+        productB = data.getReferencedProductB();
+        product1 = data.getProduct1();
+        product2 = data.getProduct2();
+    }
 
-        productSomeId = findBySku.apply(SKU_SOME_ID).orElseGet(() -> createTestProduct("Some Id", ProductVariantDraftBuilder.of().sku(SKU_SOME_ID).build()));
-        productOtherId = findBySku.apply(SKU_OTHER_ID).orElseGet(() -> createTestProduct("Other Id", ProductVariantDraftBuilder.of().sku(SKU_OTHER_ID).build()));
-        product1 = findBySku.apply(SKU1).orElseGet(() -> createProduct1());
-        product2 = findBySku.apply(SKU2).orElseGet(() -> createProduct2());
+    @AfterClass
+    public static void clean() {
+        destroy(client(), data);
```

Comment:
That's bad, because we already reached the conclusion not so long ago that with eventual consistency only keeping the products was a solution :P
Anyway if it works, then there's no problem.